### PR TITLE
M3

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Comment by sdelamo** [comment]
@wetted  there is a breaking change. Please investigate what has changed in core and fix it in servlet. 

 ```
/Users/sdelamo/github/micronaut-projects/micronaut-servlet/servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java:70: error: getAnnotationType() in ServletBodyBinder cannot override getAnnotationType() in DefaultBodyAnnotationBinder
    public Class<Body> getAnnotationType() {
                       ^
  overridden method is final
```

**Comment by graemerocher** [comment]
Seems like this method can just be deleted https://github.com/micronaut-projects/micronaut-servlet/blob/master/servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java#L70-L72

**Comment by wetted** [comment]
> Seems like this method can just be deleted https://github.com/micronaut-projects/micronaut-servlet/blob/master/servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java#L70-L72

Yeah, it can. The bigger problems are that `DefaultBodyAnnotationBinder.bind(ArgumentConversionContext<T> context, HttpRequest<?> source)` in core is now `final` and the following override it.

- https://github.com/micronaut-projects/micronaut-servlet/blob/294a81084e5433e3ac09a57ae47e9adce143b287/servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java#L75
- https://github.com/micronaut-projects/micronaut-servlet/blob/294a81084e5433e3ac09a57ae47e9adce143b287/servlet-engine/src/main/java/io/micronaut/servlet/engine/bind/DefaultServletBinderRegistry.java#L90

File: http-server-undertow/src/test/groovy/io/micronaut/servlet/undertow/UndertowParameterBindingSpec.groovy [hunk 1]
```diff
@@ -91,7 +91,7 @@ class UndertowParameterBindingSpec extends Specification {
         expect:
         response.status() == HttpStatus.BAD_REQUEST
         response.body().contains('Failed to convert argument')
-        response.body().contains('Expected one integer, but got array of multiple value')
+        response.body().contains('Expected one string, but got array of multiple value')
```
Comment (sdelamo) [inline http-server-undertow/src/test/groovy/io/micronaut/servlet/undertow/UndertowParameterBindingSpec.groovy:94]: is this change expected?

**Review by dstepanov — APPROVED** [review]

File: http-server-undertow/src/test/groovy/io/micronaut/servlet/undertow/UndertowParameterBindingSpec.groovy (hunk 1 above)
Comment (yawkat) [reply http-server-undertow/src/test/groovy/io/micronaut/servlet/undertow/UndertowParameterBindingSpec.groovy:94]: yes the old message was wrong

File: servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java [hunk 2]
```diff
@@ -166,7 +171,7 @@ public List<ConversionError> getConversionErrors() {
             }
 
         }
-        return super.bind(context, source);
+        return defaultBodyAnnotationBinder.bind(context, source);
```
Comment (yawkat) [inline servlet-core/src/main/java/io/micronaut/servlet/http/ServletBodyBinder.java:174]: would be nice to eventually replace this method entirely

File: servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java [hunk 3]
```diff
@@ -189,9 +189,9 @@ protected Object buildBody() {
     private Argument<?> resolveBodyType() {
         RouteMatch<?> route = this.getAttribute(HttpAttributes.ROUTE_MATCH, RouteMatch.class).orElse(null);
         if (route != null) {
-            Argument<?> bodyType = route.getRouteInfo().getFullBodyArgument()
+            Argument<?> bodyType = route.getRouteInfo().getFullRequestBodyType()
                 /*
-                The getBodyArgument() method returns arguments for functions where it is
```
Comment (yawkat) [inline servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java:194]: not sure this filter is necessary anymore

**Review by yawkat — APPROVED** [review]

File: servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java (hunk 3 above)
Comment (wetted) [reply servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java:194]: I removed the filter. Please confirm I did what you expected.

**Pull request merged**